`rustc --explain E0582` additional example (#124746)
rust-lang/rust: compiler/rustc_error_codes/src/error_codes/E0582.md

fn main() { }
```

This error also includes the use of associated types with lifetime parameters.
```compile_fail,E0582
trait Foo {
    type Assoc<'a>;
}

struct Bar<X, F>
where
    X: Foo,
    F: for<'a> Fn(X::Assoc<'a>) -> &'a i32
{
    x: X,
    f: F
}
```

> This is as `Foo::Assoc<'a>` could be implemented by a type that does not use

Reviewer: ```suggestion
That scenario encounters this error because `Foo::Assoc<'a>` could be implemented by a type that does not use
```
